php-checks/src/main/java/org/sonar/php/checks/OverwrittenArrayElementCheck.java

```diff
@@ -35,15 +35,27 @@
 import org.sonar.plugins.php.api.visitors.PHPSubscriptionCheck;
 import org.sonar.plugins.php.api.visitors.PHPTreeSubscriber;
 
+import static java.util.Arrays.asList;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 @Rule(key = "S4143")
 public class OverwrittenArrayElementCheck extends PHPSubscriptionCheck {
   private static final String MESSAGE = "Verify this is the array key that was intended to be written to; a value has already been saved for it and not used.";
   private static final String MESSAGE_SECONDARY = "Original assignment.";
+  static final Set<String> PREDEFINED_GLOBAL_ARRAYS = new HashSet<>(asList("$GLOBALS",
+    "$_SERVER",
+    "$_REQUEST",
+    "$_POST",
+    "$_GET",
+    "$_FILES",
+    "$_ENV",
+    "$_COOKIE",
+    "$_SESSION"));
```

Comment:
I think you could have used `CheckUtils.SUPERGLOBALS`.

Reply:
Done. Didn't see that it already existed. Thanks !